src/App.tsx

```diff
@@ -15,37 +18,96 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortField {
+  name = 'name',
+  length = 'length',
+}
+
+const SORT_FIELD_NAME: SortField = SortField.name;
+const SORT_FIELD_LENGTH: SortField = SortField.length;
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: string,
+  reverse: boolean,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return good1.localeCompare(good2);
+        case SORT_FIELD_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return reverse ? preparedGoods.reverse() : preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState('');
```

Comment:
The state 'sortField' should be initialized with a specific sort field or null to indicate no sorting. Initializing with an empty string is not clear and might lead to unexpected behavior when checking the sort field.